File: Algorithms/Sorting/Counting Sort 1/solution.py
#https://www.hackerrank.com/challenges/countingsort1/problem
#!/bin/python3

import math
import os
import random
import re
import sys

def countingSort(arr):
    count = [0] * 100
    for i in range(0,len(arr)):
        count[arr[i]]+= 1
    return count

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    arr = list(map(int, input().rstrip().split()))

    result = countingSort(arr)

    fptr.write(' '.join(map(str, result)))
    fptr.write('\n')

    fptr.close()

File: Algorithms/Sorting/Counting Sort 1/solution.js
//https://www.hackerrank.com/challenges/countingsort1/problem
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

function countingSort(arr) {
    var count = new Array(100).fill(0);
    for(var i = 0; i < arr.length; i++)
        count[arr[i]]+= 1;

    return count

}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const n = parseInt(readLine(), 10);

    const arr = readLine().split(' ').map(arrTemp => parseInt(arrTemp, 10));

    let result = countingSort(arr);

    ws.write(result.join(" ") + "\n");

    ws.end();
}